addons/road-generator/road_point.gd

```diff
@@ -456,3 +456,39 @@ func _exit_tree():
 			if singling_rp_ref != rp_ref.get_path_to(self):
 				pass
 			singling_rp_ref = null
+
+
+## Evaluates a RoadPoint's prior/next_pt_inits and verifies that they describe a
+## valid junction. A junction is valid if this RoadPoint agrees with what the
+## associated RoadPoint is saying. Invalid junctions are cleared. But, only if
+## auto_refresh is true.
+func validate_junctions(auto_refresh: bool):
+	if not auto_refresh:
+		return
+
+	var prior_point: RoadPoint
+	var next_point: RoadPoint
+	var eval_point: RoadPoint
+	var eval_pt_init: NodePath
+
+	# Get valid Prior and Next RoadPoints
+	if prior_pt_init and not prior_pt_init == "":
+		prior_point = get_node(prior_pt_init)
+	if next_pt_init and not next_pt_init == "":
+		next_point = get_node(next_pt_init)
+
+	# Verify Prior RoadPoint identifies this one as Next
+	if is_instance_valid(prior_point):
+		eval_pt_init = prior_point.next_pt_init
+		if eval_pt_init and not eval_pt_init == "":
+			eval_point = get_node(eval_pt_init)
+			if not eval_point == self:
+				prior_pt_init = null
+
+	# Verify Next RoadPoint identifies this one as Prior
+	if is_instance_valid(next_point):
+		eval_pt_init = next_point.prior_pt_init
+		if eval_pt_init and not eval_pt_init == "":
+			eval_point = get_node(eval_pt_init)
+			if not eval_point == self:
+				next_pt_init = null
```

Comment:
I realize this will complicate this, but I'm considering if we should make this a _tad_ more permissive. 

Taking the first `is_instance_valid(prior_point)` branch: This points to the connected `prior_point` roadpoint. Could we check if _either_ `prior_point`'s `next_pt_init` _or_ its `prior_pt_init` indicate being connected to the current point we are validating?

I'm imagining a scenario where someone has created a (long) dirt road, and is connecting it to another road of asphault. Or othwerise was creating roads in their world and starting at different ends and wanting to connect them. If they are facing the opposite way, this code stops this from being able to connect these two segments. 

If we want to support the user being able to connect roadpoints facing either direction, we either would have to iterate through the list of roadpoints and have flip the connection order on one of them (seems error prone), or just handle here allowing them to face either direction. Generally, facing either direction works ok now and I'd prefer that approach, especially if in the future we attach settings on how to draw segments (barriers etc), since then flipping the direction of one road to connect it to another would have to also move which roadpoitns those settings are attached to as a part of flipping the direction.

Can discuss further if needed.